.github/workflows/lint.yml

```diff
@@ -7,21 +10,31 @@ on:
       - develop
 
 jobs:
+  # Running pre-commit in CI. See semgrep/.pre-commit-config.yaml for
+  # our pre-commit configuration.
   pre-commit:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        #TODO: why do we need this ref:? doesn't checkout@v3 grabs the right branch by default?
         with:
           ref: ${{ github.event.pull_request.head.sha }}
+      # We grab those submodules below because they are the one needed by 'mypy' which
+      # runs as part of pre-commit to check our Python code.
+      # alt: we could also use 'submodules: recursive' instead, but that would be slower
       - name: Fetch semgrep-cli submodules
         run: git submodule update --init --recursive --recommend-shallow cli/src/semgrep/lang cli/src/semgrep/semgrep_interfaces
+      #TODO: why this is needed? pre-commit/action@v3 requires this setup-python to run?
+      # what is this cache-dependency-path?
       - uses: actions/setup-python@v4
         with:
           python-version: "3.10"
           cache: pip
           cache-dependency-path: .github/workflows/lint.yml
       - uses: pre-commit/action@v3.0.0
 
+  #TODO: what is the difference with the 'pre-commit' job above? looks only
+  # the extra_args below but what is it for? What will not work without that?
```

Comment:
The extra args runs https://github.com/returntocorp/semgrep/blob/develop/.pre-commit-config.yaml#L150
which dont run on normal pre-commit since they would slow down pre-commit on local development. The intention is a test that runs semgrep-pre-commit. So we can also split this out to a different config?

Reply:
why not add just the   
```    - uses: pre-commit/action@v3.0.0
        with:
          extra_args: --hook-stage manual
```
in the pre-commit job instead of duplicating all the parts before and make it a separate job?
Because it can run in parallel then?

Reply:
I don't see any difference though oin the pre-commit config file between:
```
  # Dogfood! running semgrep in pre-commit!
  - repo: https://github.com/returntocorp/semgrep
    rev: "v0.100.0"
    hooks:
      - id: semgrep
        name: Semgrep Python
        types: [python]
        exclude: "^cli/tests/.+$|^scripts/.+$|^cli/setup.py$"
        args: ["--config", "https://semgrep.dev/p/python", "--error"]
      - id: semgrep
        name: Semgrep Bandit
        types: [python]
        exclude: "^cli/tests/.+$|^scripts/.+$|^cli/setup.py$"
        args: ["--config", "https://semgrep.dev/p/bandit", "--error"]
```
and
```
  # Run develop semgrep. Only used in CI since it's slower for local developmemt.
  # To run locally use `pre-commit run --hook-stage manual semgrep-docker-develop`
  - repo: https://github.com/returntocorp/semgrep
    rev: "v0.100.0"
    hooks:
      - id: semgrep-docker-develop
        name: Semgrep Develop Python
        types: [python]
        exclude: "^cli/tests/.+$|^scripts/.+$|^cli/setup.py$"
        args: ["--config", "p/python", "--error"]
        stages: [manual]
      - id: semgrep-docker-develop
        name: Semgrep Develop Bandit
        types: [python]
        exclude: "^cli/tests/.+$|^scripts/.+$|^cli/setup.py$"
        args: ["--config", "p/bandit", "--error"]
        stages: [manual]
```
in both cases we use rev: "v0.100.0" so we don't really test the develop docker image here.